src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/__tests__/Terminal.component.test.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import TerminalView from '../Terminal.component'
+import { SocketConnectionType } from '../../node.type'
+import * as data from '../../../../../../../../components/common/helpers/Helpers'
+import { mockUseHeightObserver, renderStrip } from '../__mocks__/terminalWrapper.mock'
+import { WebSocket, Server } from 'mock-socket'
+import { BrowserRouter } from 'react-router-dom'
+
+describe('TerminalView', () => {
+    let mockServer
+    let mockSocket
+
+    beforeAll(() => {
+        mockServer = new Server('ws://localhost:8080')
+        mockSocket = new WebSocket('ws://localhost:8080')
+    })
+
+    afterAll(() => {
+        mockServer.stop()
+        mockSocket.close()
+    })
+
+    it('renders without crashing', () => {
+        const useHeightObserverMock = jest.spyOn(data, 'useHeightObserver').mockImplementation(mockUseHeightObserver)
+        const { container } = render(
+            <TerminalView
+                terminalRef={{current: null}}
+                sessionId={''}
+                socketConnection={SocketConnectionType.CONNECTING}
+                setSocketConnection={jest.fn()}
+                clearTerminal={false}
+                renderConnectionStrip={jest.fn()}
+            />,
+            {
+                wrapper: BrowserRouter,
+            },
+        )
+        expect(container).toBeInTheDocument()
+        useHeightObserverMock.mockRestore()
+    })
+
+    it('renders with messaging strip', () => {
+        const useHeightObserverMock = jest.spyOn(data, 'useHeightObserver').mockImplementation(mockUseHeightObserver)
+        const { container } = render(
+            <TerminalView
+                terminalRef={{current: null}}
+                sessionId={''}
+                socketConnection={SocketConnectionType.CONNECTING}
+                setSocketConnection={jest.fn()}
+                clearTerminal={false}
+                renderConnectionStrip={renderStrip}
+            />,
+            {
+                wrapper: BrowserRouter,
+            },
+        )
+        expect(container).toBeInTheDocument()
+        const messageStrip = container.querySelector(
+            '.terminal-strip.pl-20.pr-20.w-100.bcr-7.cn-0.connection-status-strip',
+        )
+        expect(messageStrip).toBeInTheDocument()
+        useHeightObserverMock.mockRestore()
+    })
+
+    it('displays data received on socket connection', () => {
+        mockSocket.on = jest.fn((event) => {
+            const message = event.data
+            expect(getByText(message)).toBeInTheDocument()
+        })
+
+        const useHeightObserverMock = jest.spyOn(data, 'useHeightObserver').mockImplementation(mockUseHeightObserver)
+
+        const { getByText } = render(
```

Comment:
This should be asserted somewhere? Please remove if no use

Reply:
as per dicussion we are using it so not removing